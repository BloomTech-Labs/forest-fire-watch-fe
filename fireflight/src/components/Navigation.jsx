import React, { useState, useEffect, useContext } from "react";
import { FireContext } from "../context/contextProvider";
import { withRouter, NavLink, Link } from "react-router-dom";
import styled from "styled-components";

function Navigation({
  toggleAuthForms,
  toggleLoginStatus,
  toggleRegisterStatus
}) {
  const data = useContext(FireContext);
  const [menuToggle, setMenuToggle] = useState(false);

  const logout=e=>{
    data.state.remote.logout()
  }

  return (
    <NavContainer>
      <Logo>LOGO</Logo>
      <HamburgerMenu onClick={() => setMenuToggle(!menuToggle)}>
        <HamburgerMenuBar />
        <HamburgerMenuBar />
        <HamburgerMenuBar />
        {menuToggle ? (
          <MenuContainer>
            <MenuItem>
              <NavLink to="/" activeClassName="current">
                Home
              </NavLink>
            </MenuItem>

            {localStorage.getItem('token') == null && (
              <React.Fragment>
                <MenuItem
                  onClick={() => {
                    toggleAuthForms(true);
                    toggleRegisterStatus(true);
                    toggleLoginStatus(false);
                  }}
                >
                  <p>Register</p>
                </MenuItem>
                <MenuItem
                  onClick={() => {
                    toggleAuthForms(true);
                    toggleRegisterStatus(false);
                    toggleLoginStatus(true);
                  }}
                >
                  <p>Login</p>
                </MenuItem>
              </React.Fragment>
            )}
            {localStorage.getItem('token') != null && (
              <React.Fragment>
                <MenuItem>
                  <NavLink to="/update" activeClassName="current">
                    Update
                  </NavLink>
                </MenuItem>
                <MenuItem>
                  <Link to="/" onClick={logout}>
                    Logout
                  </Link>
                </MenuItem>
                <MenuItem>
                  <NavLink to="/address" activeClassName="current">
                    Input Your Address
                  </NavLink>
                </MenuItem>
              </React.Fragment>
            )}
          </MenuContainer>
        ) : null}
      </HamburgerMenu>
    </NavContainer>
  );
}

export default Navigation;

const NavContainer = styled.div`
  width: 100%;
  height: 40px;
  display: flex;
  justify-content: space-between;
  align-items: center;
  border-bottom: 1px solid grey;
`;

const Logo = styled.h3``;

const HamburgerMenu = styled.div`
  width: 25px;
  height: auto;
  padding: 5px;
`;

const HamburgerMenuBar = styled.div`
  width: 25px;
  height: 3px;
  background-color: black;
  margin: 5px 0;
`;

const MenuContainer = styled.div`
  position: absolute;
  right: 0;
  display: flex;
  flex-direction: column;
`;

const MenuItem = styled.div`
  padding: 10px;
`;
