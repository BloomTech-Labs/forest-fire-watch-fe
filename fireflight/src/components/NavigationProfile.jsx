// import React, { useState, useEffect, useContext } from 'react'

// import { GlobalContext } from '../context/contextProvider'
// import { withRouter, NavLink, Redirect } from 'react-router-dom'

// function NavigationProfile({
//   toggleAuthForms,
//   toggleLoginStatus,
//   toggleRegisterStatus,
//   location
// }) {
//   const data = useContext(GlobalContext)
//   const [menuToggle, setMenuToggle] = useState(false)

//   useEffect(() => {
//     let w = Math.max(
//       document.documentElement.clientWidth,
//       window.innerWidth || 0
//     )
//     if (w > 0) {
//       setMenuToggle(true) //sets menuToggle to be on all the time at any size
//     }
//   }, [])

//   const logout = e => {
//     data.state.remote.logout()
//   }

//   const protec = ['/dashboard', '/address', '/maps', '/profile']

//   if (
//     localStorage.getItem('token') == null &&
//     protec.includes(location.pathname)
//   ) {
//     return <Redirect to="/" />
//   }

//   return (
//     <div className="profile-container">
//       <NavLink exact to="/" activeClassName="current">
//         <div className="profile-item" id="profile">
//           Home
//         </div>
//       </NavLink>

//       {/* {localStorage.getItem('token') == null && (
// 				<React.Fragment>
// 					<div
// 						className="profile-item inactive"
// 						onClick={() => {
// 							toggleAuthForms(true);
// 							toggleRegisterStatus(false);
// 							toggleLoginStatus(true);
// 						}}
// 					>
// 						Sign In
// 					</div>
// 					<div
// 						className="profile-item inactive"
// 						onClick={() => {
// 							toggleAuthForms(true);
// 							toggleRegisterStatus(true);
// 							toggleLoginStatus(false);
// 						}}
// 					>
// 						Sign Up
// 					</div>
// 				</React.Fragment>
// 			)} */}
//       {localStorage.getItem('token') != null && (
//         <React.Fragment>
//           <NavLink to="/dashboard" activeClassName="current">
//             <div className="profile-item" id="profile" data-temp="here">
//               Profile
//             </div>
//           </NavLink>
//           {/* <NavLink to="/address" activeClassName="current">
// 						<div className="menu-item"> Input Your Address</div>
// 					</NavLink> */}
//           {/* <div className="menu-item"> */}
//           <NavLink to="/" onClick={logout}>
//             <div className="profile-item logout" data-temp="here">
//               Logout
//             </div>
//           </NavLink>
//           {/* </div> */}
//         </React.Fragment>
//       )}
//     </div>
//   )
// }

// export default withRouter(NavigationProfile)
