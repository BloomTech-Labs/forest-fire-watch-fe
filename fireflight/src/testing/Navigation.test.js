import React from "react";
import Navigation from "../components/Navigation";
import NavigationProfile from "../components/NavigationProfile";
import { render, fireEvent } from "./test-utils";
import { createMemoryHistory } from 'history'

describe("<Navigation />", () => {

	it("renders without crashing", () => {
		const history = createMemoryHistory()
	   	render(<Navigation />)
	})
	it("renders", () => {
		const { asFragment } = render(<Navigation />)
		expect(asFragment()).toMatchSnapshot()
	})
	it("renders Home", () => {
		const nav = render(<Navigation />);
	    
		nav.getByText(/Home/i)
	})
	it("renders Sign In in demo mode", () => {
		const nav = render(<Navigation />);

		nav.getByText(/Sign In/i)
	})
	it("renders Sign Up in demo mode", () => {
		const nav = render(<Navigation />);

		nav.getByText(/Sign Up/i)
	})
});

describe("<NavigationProfile /> while logged in", () => {
	it("renders without crashing", () => {
		const history = createMemoryHistory();
		render(<NavigationProfile />) 
	})
	it("renders", () => {
		const { asFragment } = render(<Navigation />)
		expect(asFragment()).toMatchSnapshot()
	})
	it("renders Profile when logged in", () => {
		const navProf = render(<NavigationProfile />);
		// const user = {
		// 	email: 'test@testing.com',
		// 	password: 'password123'
		// }

		navProf.getByText(/Sign Up/i)
	})
	// it("renders Logout when logged in", () => {
	// 	const nav = render(<NavigationProfile />);

	// 	nav.getByText(/Logout/i)
	// })
})