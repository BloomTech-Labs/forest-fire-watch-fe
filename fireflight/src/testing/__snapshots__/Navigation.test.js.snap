// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> renders 1`] = `
<DocumentFragment>
  <div
    class="nav-container"
  >
    <a
      aria-current="page"
      class="current"
      href="/"
    >
      <div
        class="menu-item"
      >
        Home
      </div>
    </a>
    <div
      class="menu-item inactive"
    >
      Sign In
    </div>
    <div
      class="menu-item inactive"
    >
      Sign Up
    </div>
  </div>
</DocumentFragment>
`;

exports[`<NavigationProfile /> while logged in renders 1`] = `
<DocumentFragment>
  <div
    class="nav-container"
  >
    <a
      aria-current="page"
      class="current"
      href="/"
    >
      <div
        class="menu-item"
      >
        Home
      </div>
    </a>
    <div
      class="menu-item inactive"
    >
      Sign In
    </div>
    <div
      class="menu-item inactive"
    >
      Sign Up
    </div>
  </div>
</DocumentFragment>
`;
